### PR TITLE
Add install opt to set annotations on acorn-api pods

### PR DESCRIPTION
**Pull request by njhale**
Enable custom annotations to be set on the acorn-api deployment by adding an `--api-server-pod-annotations` option to the `acorn install` subcommand

e.g.

```sh
$ acorn install --api-server-pod-annotations foo=bar
```

### CODE REVIEW

File: pkg/cli/install.go [hunk 1]
```diff
@@ -107,3 +110,15 @@ func (i *Install) Run(cmd *cobra.Command, args []string) error {
 
 	return install.Install(cmd.Context(), image, opts)
 }
+
+func parseAnnotations(annotations []string) (map[string]string, error) {
+	result := map[string]string{}
```
Comment (iwilltry42) [inline pkg/cli/install.go:115]: Nit:
```suggestion
	result := make(map[string]string, len(annotations))
```

File: docs/docs/100-reference/01-command-line/acorn_install.md [hunk 2]
```diff
@@ -27,6 +27,7 @@ acorn install
       --allow-user-metadata-namespace strings           Allow these namespaces to propagate labels and annotations to dependent objects, no effect if --ignore-user-labels-and-annotations not true
       --api-server-cpu string                           The CPU to allocate to the runtime-api-server in the format of <req>:<limit> (example 200m:1000m)
       --api-server-memory string                        The memory to allocate to the runtime-api-server in the format of <req>:<limit> (example 256Mi:1Gi)
+      --api-server-pod-annotations stringArray          annotations to apply to acorn-api pods
```
Comment (iwilltry42) [inline docs/docs/100-reference/01-command-line/acorn_install.md:30]: Is there any chance that we'll need this for the controller pod as well soon?

File: pkg/cli/install.go [hunk 3]
```diff
@@ -83,22 +84,24 @@ func (i *Install) Run(cmd *cobra.Command, args []string) error {
...
 	}
 
-	annotations := map[string]string{}
-	for _, anno := range i.ControllerServiceAccountAnnotation {
-		k, v, ok := strings.Cut(anno, "=")
-		if !ok {
-			return fmt.Errorf("--controller-service-account-annotation must be in key=value format got [%s]", anno)
-		}
-		annotations[k] = v
+	controllerAnnotations, err := parseAnnotations(i.ControllerServiceAccountAnnotation)
```
Comment (iwilltry42) [inline pkg/cli/install.go:87]: Maybe `controllerSAAnnotations` for clarity?

File: pkg/install/install.go [hunk 4]
```diff
@@ -597,6 +603,29 @@ func Deployments(runtimeImage string, apiServerReplicas, controllerReplicas int,
...
 
+func replacePodAnnotations(annotations map[string]string, objs []kclient.Object) ([]kclient.Object, error) {
+	val := map[string]any{}
+	for k, v := range annotations {
+		val[k] = v
+	}
+
+	if len(val) == 0 {
+		return objs, nil
+	}
```
Comment (iwilltry42) [inline pkg/install/install.go:614]:
```suggestion
	if len(annotations) == 0 {
		return objs, nil
	}
	
	val := make(map[string]any, len(annotations))
	for k, v := range annotations {
		val[k] = v
	}
```

**Review by iwilltry42 — APPROVED** [review]
Some comments, but nothing blocking :+1:

**Review by tylerslaton — APPROVED** [review]

File: pkg/cli/install.go (hunk 3 above)
Comment (njhale) [reply pkg/cli/install.go]: done

File: docs/docs/100-reference/01-command-line/acorn_install.md (hunk 2 above)
Comment (njhale) [reply docs/docs/100-reference/01-command-line/acorn_install.md:30]: Maybe -- I think we should add it when we need it though.

File: pkg/cli/install.go (hunk 1 above)
Comment (njhale) [reply pkg/cli/install.go]: done

**Review by g-linville — APPROVED** [review]
+1 to all of Thorsten's comments, approved

**Pull request merged**